src/Tring/ConnectionTester.cs

```diff
@@ -20,46 +20,19 @@
 using System.Net.NetworkInformation;
 using System.Net.Sockets;
 using System.Linq;
-using System.Text.RegularExpressions;
-using System.Threading.Tasks;
 
 namespace Tring
 {
     internal class ConnectionTester
     {
         private readonly TimeSpan waitTime = TimeSpan.FromSeconds(1);
-        
-        public static readonly Regex SplitFormat = new Regex(@"^(?<host>.*):(?<port>\w+)$");
+
         public enum ConnectionStatus { Succes, TimeOut, Refused, Untried };
-        public ConnectionRequest request;
+        private ConnectionRequest _request;
```

Comment:
The request can currently not be readonly because if it contains a URL the IP found by the DNS lookup will create a new request with the IP included and overwrite the old one.